mbed_targets/mbed_targets.py

```diff
@@ -52,23 +55,17 @@ class MbedTargets:
     """Interface to the Online Target Database."""
 
     def __init__(self) -> None:
-        """Retrieve target data from the online database."""
-        self._target_data: Optional[
-            List[dict]] = target_database.get_target_data()
-
-    def __iter__(self) -> 'MbedTargets':
-        """Return the iterator object itself."""
-        # Create a new iterator
-        self._target_iterator: Optional[Iterator] = iter(
-            self._target_data) if self._target_data else None
-        return self
-
-    def __next__(self) -> Optional[MbedTarget]:
-        """Return the next item from the container.
-
-        Returns:
-            An instance of a Mbed target database entry.
+        """Retrieve target data from the online database.
+
+        Raises:
+            MbedTargetDataError: failed to get data from the target database.
         """
-        # Iterate over the target data and return a new MbedTarget instance for each database entry
-        return MbedTarget(
-            next(self._target_iterator)) if self._target_iterator else None
+        try:
+            self._target_data: List[dict] = target_database.get_target_data()
+        except target_database.TargetDatabaseError as err:
```

Comment:
I think we can let this exception go and just indicate that this method raises a  `ToolsError`

Reply:
I'd prefer not to catch such a general exception here. This class has a specific invariant, which is that we can actually get some data from the target database. Catching the `TargetDataBase` error communicates this to the caller. Catching general `ToolsError`s everywhere is almost as bad as catching a base `Exception`.

Reply:
I meant we shouldn't catch the exception here at all, in the previous cases we caught a general exception and raised one based on `ToolsError` which is helpful to the user. I think whatever exception is raised in `get_target_data` can be passed back to the caller of this class. Raising a slightly different tools exception doesn't seem to add anything.

Reply:
Yeah, when you put it like that I think I agree, re-raising the slightly different `ToolsError` is a bit redundant. I'll make the change in the next commit.